contrib/setup-dashd.py

```
@@ -0,0 +1,168 @@
+    return asset
+
+
+def log(msg):
+    print(msg, file=sys.stderr)
+
+
+def download(url, dest, timeout=300, retries=3):
+    for attempt in range(1, retries + 1):
+        try:
+            log(f"Downloading {url} (attempt {attempt}/{retries})...")
+            with urllib.request.urlopen(url, timeout=timeout) as response:
```

Ruff (0.15.6)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)